atom/browser/common_web_contents_delegate.cc

```diff
@@ -120,16 +125,18 @@ PrefService* GetPrefService(content::WebContents* web_contents) {
   return static_cast<atom::AtomBrowserContext*>(context)->prefs();
 }
 
-std::set<std::string> GetAddedFileSystemPaths(
+std::map<std::string, std::string> GetAddedFileSystemPaths(
     content::WebContents* web_contents) {
   auto* pref_service = GetPrefService(web_contents);
   const base::DictionaryValue* file_system_paths_value =
       pref_service->GetDictionary(prefs::kDevToolsFileSystemPaths);
-  std::set<std::string> result;
+  std::map<std::string, std::string> result;
   if (file_system_paths_value) {
     base::DictionaryValue::Iterator it(*file_system_paths_value);
     for (; !it.IsAtEnd(); it.Advance()) {
-      result.insert(it.key());
+      std::string type =
+          it.value().is_string() ? it.value().GetString() : std::string();
+      result[it.key()] = type;
```

Comment:
Not sure I understand the desired result here. Why do we want some entries in the map to be ""? In what conditions do we get an `it.value()` that's not a string?

Reply:
For each loaded file path devtools expects an associated file system type, when its not special an empty value must be provided.

>  In what conditions do we get an it.value() that's not a string?

`base::Value::Get*` have `CHECK`, to avoid a crash in release build due to some invalid values, the check is present as a guard. As the type value is something that is generated by the devtools rather than by a user.